libraries/botbuilder-dialogs-adaptive-runtime/src/index.ts

```diff
@@ -184,7 +184,17 @@ function addStorage(services: ServiceCollection, configuration: Configuration):
 }
 
 function addSkills(services: ServiceCollection, configuration: Configuration): void {
-    services.addInstance('credentialProvider', new SimpleCredentialProvider('appId', 'appPassword'));
+    const appId = configuration.string(['MicrosoftAppId']);
+    if (!appId) {
+        throw new Error('`MicrosoftAppId` not found in configuration');
+    }
+
+    const appPassword = configuration.string(['MicrosoftAppPassword']);
+    if (!appPassword) {
+        throw new Error('`MicrosoftAppPassword` not found in configuration');
+    }
```

Comment:
This is for both addSkills and the addFactory call below:

I think these checks are too stringent, the SDK permits skills without AppIds. Similarly users can create a Composer bot without having to make an AAD app registration and Azure Bot Resource. And can a credentialProvider have already been passed in?

I think the logic might look like this:
```
If a credential provider exists, use that or,
If an appId and appPassword exists, create a new credentialProvider or,
Create an empty credential provider for no auth
```

Reply:
I think we need to respect the existing `BotFrameworkAdapter` constructor until we replace it with `CloudAdapter`. That said, I'm no longer throwing errors which _I think_ aligns with the behavior you suggest.

Reply:
Yes, the behavior around missing appIds etc was my main concern. Thanks!